lib/controller/api/ForumController.class.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+use \Jacwright\RestServer\RestException;
+
+class ApiForumController {
+
+	/**
+	 * @return boolean
+	 */
+	public function authorize() {
+		return ApiAuthController::isAuthorized() && LoginModel::mag('P_OUDLEDEN_READ');
+	}
+
+	/**
+	 * @url GET /recent
+	 * @param int offset
+	 * @param int limit
+	 * @return ForumDraad[]
+	 */
+	public function getRecent() {
+		$offset = isset($_GET['offset']) ? intval($_GET['offset']) : 0;
+		$limit = isset($_GET['limit']) ? intval($_GET['limit']) : 10;
+
+		$draden = ForumDradenModel::instance()->getRecenteForumDraden($limit, null, false, $offset, true);
+
+		foreach ($draden as $draad) {
+			$draad->ongelezen = $draad->isOngelezen();
+			$draad->laatste_post = ForumPostsModel::instance()->get($draad->laatste_post_id);
+			$draad->laatste_wijziging_naam = ProfielModel::getNaam($draad->laatste_wijziging_uid, 'civitas');
+		}
+
+		return array('data' => array_values($draden));
+	}
+
+	/**
+	 * @url GET /onderwerp/$id
+	 * @param int offset
+	 * @param int limit
+	 * @return ForumPost[]
+	 */
+	public function getOnderwerp($id) {
+		$offset = isset($_GET['offset']) ? intval($_GET['offset']) : 0;
```

Comment:
Hier gebruiken we meestal `filter_input`

Hier zou bijvoorbeeld het volgende kunnen doen:

```PHP
$offset = filter_input(INPUT_GET, 'offset', FILTER_VALIDATE_INT) ?: 0;
```